### PR TITLE
fix(users): deprecate all existing API, use rxjs

### PR DESCRIPTION
**Pull request by pmuir**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -133,6 +133,7 @@
     "karma-phantomjs-launcher": "1.0.2",
     "karma-sourcemap-loader": "0.3.7",
     "karma-webpack": "2.0.2",
+    "lodash": "4.17.4",
```
Comment (joshuawilson) [inline package.json]: we already have lodash in the dependency section

File: src/app/user/user.service.ts [hunk 2]
```diff
@@ -1,107 +1,144 @@
...
   private userUrl: string;  // URL to web api
   private usersUrl: string;  // URL to web api
 
-
   constructor(private http: Http,
-              private logger: Logger,
-              private auth: AuthenticationService,
-              private broadcaster: Broadcaster,
-              @Inject(AUTH_API_URL) apiUrl: string
+    private logger: Logger,
```
Comment (joshuawilson) [inline src/app/user/user.service.ts]: if you are going to move things over, you should take the one above it too ;)

File: src/app/user/user.service.ts [hunk 3]
```diff
@@ -1,107 +1,144 @@
...
+
+  /**
+   * The currently logged in user
+   */
+  get loggedInUser(): Observable<User> {
+    return this.http
+      .get(this.userUrl, { headers: this.headers })
+      .map(response => {
+        let userData = cloneDeep(response.json().data as User);
+        userData.attributes.primaryEmail = userData.attributes.email;
```
Comment (joshuawilson) [inline src/app/user/user.service.ts]: Why are you copying the email to the primaryEmail? That feels like something the backend should do. Or we should point to the regular email.

**Comment by pmuir** [comment]
Agreed - can you file a bug as I was trying not to change the API here,
just get the right methods available for now.

File: package.json (hunk 1 above)
Comment (pmuir) [inline package.json]: Only the typings I think

**Pull request merged**